### ISSUE TITLE
It's no longer possible to "alias" the model attribute name using mutators

### ISSUE THREAD

**Issue by boryn** [laravel/framework#46431]
- Laravel Version: ^9|^10
- PHP Version: 8.1
- Database Driver & Version:

### Description:
Using "old" way of setting mutators, it was possible to "alias" an attribute name. 
With the new way, it's not possible (the accessor works, but not the mutator). 

Let's say we have the model Booking. It has the real field `start` in the database (it's in `$fillable`). The attribute `arrival` is "virtual". 

### Steps To Reproduce:

The working "old" solution, definitions at the Booking model:
```php
public function getArrivalAttribute()
{
    return $this->attributes['start'];
}

public function setArrivalAttribute($value)
{
    $this->attributes['start'] = $value;
}
```

With this old solution, this works as expected (in the database it's the `start` field which gets updated):
```php
$booking = Booking::find(1);
$booking->arrival = '2023-03-12';
$booking->save();
```

The new way definition:
```php
protected function arrival(): Attribute
{
    return Attribute::make(
        get: fn ($value, array $attributes) => $attributes['start'],
        set: function ($value): void {
            $this->attributes['start'] = $value;
        }
    );
}
```

and now, after issuing:
```php
$booking = Booking::find(1);
$booking->arrival = '2023-03-12';
$booking->save();
```

unfortunately it raises an error:
`SQLSTATE[42S22]: Column not found: 1054 Unknown column 'arrival' in 'field list' (Connection: mysql, SQL: update bookings set arrival = ?`

Tried as well to `unset($this->attributes['arrival'])` or `unset($this->arrival)` in the set method, but it does not help.

**Comment by ankurk91** [comment]
You can still keep using good old way.

**Comment by boryn** [comment]
Yes, I know, but would like to learn how to use the new way. And what when the old way gets deprecated?

**Comment by driesvints** [comment]
Thanks. We have no plans right now to remove the old way so you can keep using it if you need this.

**Comment by coolcodemy** [comment]
Facing same issue today. You could do this:

```php
protected function arrival(): Attribute
{
    return Attribute::make(
        get: fn ($value, array $attributes) => $attributes['start'],
        set: function ($value): void {
            return ['start' => $value];
        }
    );
}
```

**Comment by str** [comment]
> Thanks. We have no plans right now to remove the old way so you can keep using it if you need this.

For someone that only reads the latest documentation page, there is no way to know the "old way".
